launcher/internal/signaturediscovery/client.go

```diff
@@ -37,6 +41,36 @@ func (c *Client) FetchSignedImageManifest(ctx context.Context, targetRepository
 	return getManifest(ctx, image)
 }
 
+// FetchImageSignatures returns a list of valid image signatures associated with the target OCI image.
+func (c *Client) FetchImageSignatures(ctx context.Context, targetRepository string) ([]oci.Signature, error) {
```

Comment:
This looks good, but I don't have a lot of experience with the OCI implementation in Cosign. To confirm this is working as expected, I would either manually compare to what Cosign generates, or use this library to verify what Cosign generates.

Reply:
Done, see tests here https://github.com/google/go-tpm-tools/pull/328/files#diff-fc42aca0ab2213e764bddeab995045792c9d118a151ea446c7ad03fe21b1444fR61.

Reply:
Looks great, thanks!